### PR TITLE
Machine readable output changes (global changes + error output)

### PR DESCRIPTION
**Pull request by cdrage**
This PR:
  - Adds `-o json` as a global parameter
  - Makes sure that NO logging is done when using `-o json` with the
  exception of error out / success output via json

### CODE REVIEW

File: pkg/machineoutput/types.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package machineoutput
+
+import (
+	"encoding/json"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const errorKind = "Error"
```
Comment (surajnarwade) [inline pkg/machineoutput/types.go]: also present in `pkg/odo/util/cmdutils.go`

File: pkg/machineoutput/types.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	}
+
+	// We print and then exit making sure we exit with an error code..
+	fmt.Println(string(out))
+
+	return nil
+}
+
+// Return success result to fmt.Println
+func SuccessMachineOutput(namespace string, outputMessage string, kind string) error {
```
Comment (surajnarwade) [inline pkg/machineoutput/types.go]: also present in `pkg/odo/util/cmdutils.go`

**Comment by cdrage** [comment]
Close in favour of #1898

**Pull request closed without merging**